tests/e2e/vsan_stretched_cluster_utils.go

```diff
@@ -246,3 +247,144 @@ func createPvcInParallel(client clientset.Interface, namespace string, diskSize
 		lock.Unlock()
 	}
 }
+
+// waitForPodsToBeInErrorOrRunning polls for pod to be in error or running state
+func waitForPodsToBeInErrorOrRunning(c clientset.Interface, podName, namespace string, timeout time.Duration) error {
+	waitErr := wait.PollImmediate(poll, timeout, func() (bool, error) {
+		pod, err := c.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		framework.Logf("Pod is in phase: %v", pod.Status.Phase)
+		switch pod.Status.Phase {
+		// v1.PodSucceeded is for pods in ExitCode:0 state.
+		// Standalone pods are in ExitCode:0 or Running state after site failure.
+		case v1.PodRunning, v1.PodSucceeded:
+			framework.Logf("Pod %v is in state %v", podName, pod.Status.Phase)
+			return true, nil
+		}
+		return false, nil
+	})
+	return waitErr
+}
+
+// updatePvcLabelsInParallel updates the labels of pvc in a namespace in parallel
+func updatePvcLabelsInParallel(ctx context.Context, client clientset.Interface, namespace string,
+	labels map[string]string, pvclaims []*v1.PersistentVolumeClaim, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for _, pvc := range pvclaims {
+		framework.Logf(fmt.Sprintf("Updating labels %+v for pvc %s in namespace %s",
+			labels, pvc.Name, namespace))
+		pvc, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvc.Labels = labels
+		_, err = client.CoreV1().PersistentVolumeClaims(namespace).Update(ctx, pvc, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+// updatePvLabelsInParallel updates the labels of pv in parallel
+func updatePvLabelsInParallel(ctx context.Context, client clientset.Interface, namespace string,
+	labels map[string]string, persistentVolumes []*v1.PersistentVolume, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for _, pv := range persistentVolumes {
+		framework.Logf(fmt.Sprintf("Updating labels %+v for pv %s in namespace %s",
+			labels, pv.Name, namespace))
+		pv, err := client.CoreV1().PersistentVolumes().Get(ctx, pv.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pv.Labels = labels
+		_, err = client.CoreV1().PersistentVolumes().Update(ctx, pv, metav1.UpdateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+// getMasterIpOnSite returns IP address of master node on a particular site
+func getMasterIpOnSite(ctx context.Context, client clientset.Interface, primarySite bool) (string, error) {
```

Comment:
this function should be used only for distributed model right? please add a comment indicating it.